disnake/ext/commands/interaction_bot_base.py

```diff
@@ -1275,24 +1483,29 @@ async def process_application_commands(
                 return
 
         command_type = interaction.data.type
-        command_name = interaction.data.name
         app_command = None
         event_name = None
 
+        for command in self._all_app_commands.values():
+            if command.body.id == interaction.data.id:
+                app_command = command
+                break
+            elif command.body.id is None:
+                ...
+        else:
+            app_command = None
```

Comment:
I think here we can just get command by metadata because it is provided in the interaction payload and it will work regardless of `_command_sync` flags. Also it seems like discord added `guild_id` to to the `ApplicationCommandInteractionData` payload, so we can use it and save 1 lookup (see line 1463, we had to make 2 lookups - with guild id and without guild id, now we can make just one)
```python
# assuming that guild_id is implemented in ApplicationCommandInteractionData
meta = ApplicationCommandMetadata(
    name=interaction.data.name,
    type=interaction.data.type,
    guild_id=interaction.data.guild_id,
)
```